app/controllers/partyController.js

```diff
@@ -40,6 +41,43 @@ class PartyController {
       data,
     });
   }
+
+  static editParty(req, res) {
+    const id = parseInt(req.params.id, 10);
+    let foundParty;
+    let itemIndex;
+    partydb.map((data, index) => {
+      if (data.id === id) {
+        foundParty = data;
+        itemIndex = index;
+      }
+    });
+    if (!foundParty) {
+      return res.status(404).json({
+        status: 404,
+        error: 'Party record cannot be found!',
+      });
+    }
+    if (!req.body.name) {
+      return res.status(400).json({
+        status: 400,
+        error: 'name field is required!',
+      });
+    }
+    const data = {
+      id: foundParty.id,
+      name: req.body.name || foundParty.name,
+      hqAddress: foundParty.hqAddress,
+      logoUrl: foundParty.logoUrl,
+    };
+    partydb.splice(itemIndex, 1, data);
+
+    return res.status(201).json({
+      message: 'Party name was successfully edited!',
```

Comment:
Confirm the status code for updating, and your success message is specific to only the name, would some only be able to update a name and nothing else. So the party hq address cannot change ever.

Reply:
ok